Tasks/SqlDacpacDeploymentOnMachineGroup/GenerateSqlBatchFiles.ps1

```diff
@@ -0,0 +1,215 @@
+﻿function Get-LinestoAppend
+{
+    param
+    (
+        [string]$line
+    )
+
+    $line = $line.Trim()
+    $length = $line.Length
+    $commentsIndex = $line.IndexOf("--")
+
+    #Don't split if entire line is a comment
+    if ($commentsIndex -eq 0)
+    {
+        return $line, $null
+    }
+
+    #Split if the line has only GO
+    if ($line -contains "GO")
+    {
+        $lineArr = $line -split "GO", 2
+        return $lineArr[0], $lineArr[1]
+    }
+
+    #Split if line starts with GO
+    #This avoids 'select * from goair join table2' even though there is matching pattern 'go '
+    $lineSubstring = $line.Substring(0, 3)
+    if ($lineSubstring -match "GO ")
+    {
+        $lineArr = $line -split "GO ", 2
+        return $lineArr[0], $lineArr[1]
+    }
+    
+    #Split line if it has a GO in between like 'select * from table GO create procedure'
+    $goIndex = $line.IndexOf(" GO ", [System.StringComparison]::CurrentCultureIgnoreCase)
+    if (($commentsIndex -ne -1) -and ($goIndex -ne -1) -and ($commentsIndex -lt $goIndex))
+    {
+        return $line, $null
+    }
+    if ($goIndex -ge 0)
+    {
+        $lineArr = $line -split " GO ", 2
+        return $lineArr[0], $lineArr[1]
+    }
+
+    #Split if line ends with GO
+    #This avoids 'select * from indigo' even if there is a matching pattern ending with 'go'
+    $lastCharsIndex = [int]$length - 3
+    if (($commentsIndex -ne -1) -and ($lastCharsIndex -ge 0) -and ($commentsIndex -lt $lastCharsIndex))
+    {
+        return $line, $null
+    }
+    if ($length -ge 3)
+    {
+        $lineSubstring = $line.Substring($lastCharsIndex, 3)
+        if ($lineSubstring -match " GO")
+        {
+            return $line.Substring(0, [int]$length - 3), $null
+        }
+    }
+       
+    #Send the line as is       
+    return $line, $null
+}
+
+function Test-IsSplitRequired
+{
+    param
+    (
```

Comment:
Get-LinestoAppend and Test-IsSplitRequired have duplication of logic. Can Test-IsSplitRequired be dependent on Get-LinestoAppend